zaza/openstack/charm_tests/cinder_netapp/tests.py

```diff
@@ -55,23 +65,28 @@ def test_cinder_config(self):
             expected_contents,
             timeout=2)
 
+    def _create_volumes(self, n):
+        """Create N volumes."""
+        for _ in range(n):
+            name = "zaza{}".format(uuid.uuid1().fields[0])
+            vol = self.cinder_client.volumes.create(
+                name=name,
+                size='1')
+            vol.reset_state('available')
+            self.zaza_volumes.append(vol)
+
     def test_create_volume(self):
-        """Test creating a volume with basic configuration."""
-        test_vol_name = "zaza{}".format(uuid.uuid1().fields[0])
-        vol_new = self.cinder_client.volumes.create(
-            name=test_vol_name,
-            size='1')
-        try:
+        """Test creating volumes with basic configuration."""
+        self._create_volumes(2)
+        for vol in self.cinder_client.volumes.list():
```

Comment:
Shouldn't there be a check that the volume is one of the `self.zaza_volumes`?  Otherwise, this test may be polluted by other volumes that the cinder client can see (say due to other tests run, or previous tests not cleaning up properly)?

Reply:
Good point. I'll go with `self.zaza_volumes` since it covers that specific case (I actually added it for that purpose, just forgot to use it here).